crates/futures/src/queue.rs

```diff
@@ -0,0 +1,90 @@
+use js_sys::Promise;
+use std::cell::{Cell, RefCell};
+use std::collections::VecDeque;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+
+pub(crate) trait Task {
```

Comment:
I would personally prefer to remove this `trait` and just use a type from either the singlethread/multithread modules, sort of like how `std::sys` works where it just has the same structure. The trait otherwise just sort of adds unnecessary complication.

Reply:
(there's also only ever one instantiation ever)

Reply:
I used a trait so we could reuse it in other situations, like the eventual multi-threading implementation. It could also be useful to allow users to use it as a fast Rust `queueMicrotask` polyfill.

But I guess the multi-threading implementation will need a different queue anyways, so I'm fine with changing this.